pkg/cloudprovider/providers/azure/azure_blobDiskController.go

```diff
@@ -201,15 +207,20 @@ func (c *BlobDiskController) createVHDBlobDisk(blobClient azstorage.BlobStorageC
 	// add VHD signature to the blob
 	h, err := createVHDHeader(uint64(size))
 	if err != nil {
-		blob.DeleteIfExists(nil)
+		mc := newMetricContext("blob", "delete_if_exists", c.common.resourceGroup, c.common.subscriptionID)
+		_, delErr := blob.DeleteIfExists(nil)
+		mc.Observe(delErr)
 		return "", "", fmt.Errorf("failed to create vhd header, err: %v", err)
 	}
 
 	blobRange := azstorage.BlobRange{
 		Start: uint64(size),
 		End:   uint64(vhdSize - 1),
 	}
-	if err = blob.WriteRange(blobRange, bytes.NewBuffer(h[:vhd.VHD_HEADER_SIZE]), nil); err != nil {
+	mc = newMetricContext("blob", "write_range", c.common.resourceGroup, c.common.subscriptionID)
```

Comment:
is it too detail to put every operation to metrics? IMO, whole `createVHDBlobDisk` is enough for one metric.

Reply:
@andyzhangx  This make debugging easier on operations side. One can see immediately which API call is slow or throwing errors. It is also helpful for performance investigations.

It's possible to aggregate multiple API calls into one macro-operations (e.g. createVHDBlobDisk) if a third dimension is introduced in the metrics definitions.